static/app/views/performance/transactionSummary/transactionSpans/opsFilter.tsx

```diff
@@ -0,0 +1,188 @@
+import {MutableSearch} from 'app/utils/tokenizeSearch';
+
+type Props = {
+  location: Location;
+  eventView: EventView;
+  organization: Organization;
+  handleOpChange: (op: string | undefined) => void;
+  transactionName: string;
+};
+
+export default function OpsFilter(props: Props) {
+  const {location, eventView, organization, handleOpChange, transactionName} = props;
+
+  // clear out the query string from the event view
+  // as we want to restrict queries to the op names
+  const conditions = new MutableSearch('');
+  conditions
+    .setFilterValues('event.type', ['transaction'])
+    .setFilterValues('transaction', [transactionName]);
+  const opsFilterEventView = eventView.clone();
+  opsFilterEventView.query = conditions.formatString();
+
+  const currentOp = decodeScalar(location.query.spanOp);
+
+  return (
+    <DropdownControl
+      menuWidth="240px"
+      blendWithActor
+      button={({isOpen, getActorProps}) => (
+        <DropdownButton
+          data-test-id="ops-filter-button"
+          {...getActorProps()}
+          showChevron={false}
+          isOpen={isOpen}
+        >
+          <Fragment>
+            <IconFilter size="xs" />
+            <FilterLabel>{t('Filter')}</FilterLabel>
+          </Fragment>
+        </DropdownButton>
+      )}
+    >
+      <List>
+        <ListHeader
+          data-test-id="span-op-filter-header"
+          onClick={event => {
+            event.stopPropagation();
+            handleOpChange(undefined);
+          }}
+        >
+          <HeaderTitle>{t('Operations')}</HeaderTitle>
+          <Radio radioSize="small" checked={!defined(currentOp)} onChange={() => {}} />
+        </ListHeader>
+        <SpanOpsQuery
+          location={location}
+          orgSlug={organization.slug}
+          eventView={opsFilterEventView}
+        >
+          {({spanOps, isLoading, error}) => {
+            if (isLoading) {
+              return <StyledLoadingIndicator />;
+            }
+
+            if (error) {
+              return (
+                <ErrorPanel height="124px">
+                  <IconWarning color="gray300" size="lg" />
+                </ErrorPanel>
+              );
+            }
+
+            if (!defined(spanOps) || spanOps.length === 0) {
+              return <EmptyStateWarning small>{t('No span ops')}</EmptyStateWarning>;
+            }
+
+            return spanOps.map(spanOp => (
+              <ListItem
+                data-test-id="span-op-filter-item"
+                key={spanOp.op}
+                onClick={event => {
+                  event.stopPropagation();
+                  handleOpChange(spanOp.op);
+                }}
+              >
+                <OperationDot backgroundColor={pickBarColor(spanOp.op)} />
+                <OperationName>{spanOp.op}</OperationName>
+                <Radio
+                  radioSize="small"
+                  checked={spanOp.op === currentOp}
+                  onChange={() => {}}
```

Comment:
Hmm I didn't know we would do a fake radio and make the whole row clickable but that's probably better considering a dropdown is ephemeral. I'm guessing we do this elsewhere?

Reply:
We actually have this pattern for the [ops breakdown filter](https://github.com/getsentry/sentry/blob/master/static/app/views/performance/transactionSummary/filter.tsx#L112-L123) already. I had to add this empty `onChange` handler to make a warning go away about the radio button missing a change handler.

Reply:
Yeah it's fine, the empty onChange is what caught my eye, radios can have small click areas anyway.